core-api/src/main/java/com/optimizely/ab/notification/DecisionNotification.java

```diff
@@ -0,0 +1,167 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package com.optimizely.ab.notification;
+
+
+import com.optimizely.ab.bucketing.FeatureDecision;
+import com.optimizely.ab.config.FeatureVariable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DecisionNotification {
+    protected String type;
+    protected String userId;
+    protected Map<String, ?> attributes;
+    protected Map<String, ?> decisionInfo;
+
+    protected DecisionNotification() {
+    }
+
+    protected DecisionNotification(String type,
+                                   String userId,
+                                   Map<String, ?> attributes,
+                                   Map<String, ?> decisionInfo) {
+        this.type = type;
+        this.userId = userId;
+        this.attributes = attributes;
+        this.decisionInfo = decisionInfo;
```

Comment:
Are all of these fields allowed to be null? I see in `BaseDecisionNotificationBuilder`, that we default `attributes` to empty map -- that should be moved here (otherwise assert non-null in constructor) to make it consistent, otherwise nulls can slip in.